File: C/Algorithms/226.c
#include <stdlib.h>

// Definition for a binary tree node.
struct TreeNode {
  int val;
  struct TreeNode *left;
  struct TreeNode *right;
};

void invert(struct TreeNode *node) {
  // Base case: leaf, simply return.
  if (!node)
    return;

  // Do the switch and recurse on children.
  struct TreeNode *tmp = node->right;
  node->right = node->left;
  node->left = tmp;
  return invert(node->left), invert(node->right);
}

struct TreeNode *invertTree(struct TreeNode *root) {
  if (!root)
    return NULL;
  invert(root);
  return root;
}

File: Javascript/Algorithms/226.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
let invert = (node) => {
    if (!node){
        return;
    }
    let tmp = node.right;
    node.right = node.left;
    node.left = tmp;
    return invert(node.left), invert(node.right);
}

var invertTree = function(root) {
    if (!root){
        return null;
    }
    invert(root);
    return root;
};
